plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/machine/MachineManagerImpl.java

```diff
@@ -189,11 +190,11 @@ public void onMachineRunning(MachineStateEvent event) {
             @Override
             public void onMachineDestroyed(MachineStateEvent event) {
                 if (isMachineRestarting) {
-                    final String recipeUrl = machineState.getConfig().getSource().getLocation();
+                    final MachineSource machineSource = machineState.getConfig().getSource();
                     final String displayName = machineState.getConfig().getName();
                     final boolean isDev = machineState.getConfig().isDev();
 
-                    startMachine(recipeUrl, displayName, isDev, RESTART, "dockerfile", "docker");
+                    startMachine(asDto(machineSource), displayName, isDev, RESTART, "dockerfile", "docker");
```

Comment:
hmm what I don't see is why it's hardcoded "dockerfile" and "docker" in machine manager ?